packages/fleet/lib/models/deployments.ts

```diff
@@ -53,19 +56,25 @@ export async function create(db: knex.Knex, commitId: CommitHash): Promise<Resul
                 })
                 .update({ superseded_at: now });
             // insert new deployment
+            const commitId = image.split(':')[1];
+            if (!commitId || commitId.length !== 40) {
+                return Err(new Error(`Error: invalid image '${image}'`));
+            }
             const dbDeployment: Omit<DBDeployment, 'id'> = {
-                commit_id: commitId,
+                commit_id: commitId as CommitHash,
+                image,
                 created_at: now,
                 superseded_at: null
             };
             const [inserted] = await trx.into<DBDeployment>(DEPLOYMENTS_TABLE).insert(dbDeployment).returning('*');
             if (!inserted) {
-                return Err(new Error(`Error: no deployment '${commitId}' created`));
+                return Err(new Error(`Error: no deployment for '${image}' created`));
             }
             return Ok(DBDeployment.to(inserted));
         });
     } catch (err) {
-        return Err(new FleetError(`deployment_creation_failed`, { cause: err, context: { commitId } }));
+        console.log(err);
```

Comment:
Not sure if this was intended or a leftover

Reply:
🤦